### ISSUE TITLE
Fix C99 bool type

### ISSUE THREAD

**Issue by SoapGentoo** [mdadams/jasper#27]
In Gentoo, we're currently applying a patch to fix the broken bool handling:
https://github.com/gentoo/gentoo/blob/master/media-libs/jasper/files/jasper-1.900.3-remove-stdbool-checks.patch

The issued caused is documented here:
https://bugs.gentoo.org/show_bug.cgi?id=513240

I suggest getting rid of all the test stuff and just asking for C99 support unconditionally. As a second solution, if you'd like to keep it the way it is, I think moving the `#undef`'s just before the `#include <stdbool.h>` should also do the job.

**Comment by mdadams** [comment]
Thanks for the explanation of the problem.  I agree with your assessment.  It's 2016.  If someone does not at least have a C99-compliant compiler, then it is time to upgrade their compiler.  I ditched all of the tests for the existence of C99 headers.  Now, the code just assumes the headers exist. I left the actual include directives for the C99 headers there for now (until they have been added to the other source files as needed).  See commit 5971949c0c88281c7e5044c367a9d03597fca3fd.
